src/components/AppWrapper.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react'
+import {IInfographicsSection, TInfographics, IInfographicsRow} from '../types/IInfographicsSection'
+import Editor from './Editor';
+import Canvas from './Canvas'
+import {IInfographicsDetails} from '../types/IInfographicsDetails'
+import produce from 'immer'
+
+interface IState {
+    infographicsDetails: IInfographicsDetails,
+    lastUsedId: number,
+    selectedSection: IInfographicsSection | null,
+    columnsCount: number,
+}
+
+class AppWrapper extends React.Component<{}, IState> {
+
+    state: IState = {
+        infographicsDetails: {
+            widthPx: 400, // the default width of infographics could be set in "New infographics" form in the future
+            infographics: [] as TInfographics,
+        },
+        lastUsedId: 0,
```

Comment:
`lastUsedId` je defakto 

`items[items.length].id + 1`, cili ID posledniho itemu + 1

ale pokud by jsi to chtela delat takto, doporucuji local variable

`lastUsedId: number = 0`

a potom k ni pristupovat jako
`this.lastUsedId` a `this.lastUsedId = 1`

pokud neni jasne jak to myslim, tak ti to vysvetlim zitra :) kdyz to mas ve `state` tak pokud to zmenis, zpusobi to re-render, ktery ale pro takovouhle informaci neni treba, lokalni variable nezpusobi re-render